### PR TITLE
fix(ci): copy package json files separately

### PR DESCRIPTION
**Pull request by clemsos**
# Description

To prevent the `COPY . .` to invalidate docker cache on each file change, we copy package.json files separately (using rsync) and build packages.

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — APPROVED** [review]
Did you try to run this locally twice? If so was the second one using the cache? Ideally please try aussi to modify a random file and make sure the cached dependencies remain cached?

**Review by julien51 — CHANGES_REQUESTED** [review]
I tried running locally and this is not successfully run:

```
------                                                                                                                                                                                                                                           
 > [dev 4/8] RUN --mount=type=bind,target=/docker-context     rsync -amv --delete           --exclude='node_modules'           --exclude='*/node_modules'           --include='package.json'           --include='*/' --exclude='*'           /docker-context/ /home/unlock;:                                                                                                                                                                                                                     
#0 0.976 building file list ...                                                                                                                                                                                                                  
#0 1.000 rsync: [sender] opendir "/docker-context/wedlocks/.netlify/functions/handler" failed: Permission denied (13)                                                                                                                            
#0 1.006 done
#0 1.010 IO error encountered -- skipping file deletion
#0 1.010 ./
#0 1.186 
#0 1.186 rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1333) [sender=3.2.3]
#0 1.186 sent 14,415 bytes  received 70 bytes  28,970.00 bytes/sec
#0 1.186 total size is 41,790  speedup is 2.89
------
failed to solve: executor failed running [/bin/sh -c rsync -amv --delete           --exclude='node_modules'           --exclude='*/node_modules'           --include='package.json'           --include='*/' --exclude='*'           /docker-context/ $DEST_FOLDER;]: exit code: 23
```

**Comment by julien51** [comment]
wait I think the `/docker-context/wedlocks/.netlify` are transient files?

**Comment by julien51** [comment]
Well I actually have more issues:

```
 > [dev 4/8] RUN --mount=type=bind,target=/docker-context     rsync -amv --delete           --exclude='node_modules'           --exclude='*/node_modules'           --include='package.json'           --include='*/' --exclude='*'           /docker-context/ /home/unlock;:                                                                                                                                                                                                                     
#0 0.276 building file list ... done                                                                                                                                                                                                             
#0 0.959 rsync: [generator] delete_file: rmdir(${DEST_FOLDER}/yarn-cache) failed: Permission denied (13)                                                                                                                                         
#0 0.960 cannot delete non-empty directory: yarn-cache                                                                                                                                                                                           
#0 0.960 deleting wiki/
#0 0.960 cannot delete non-empty directory: scripts/circleci
#0 0.960 cannot delete non-empty directory: scripts/circleci
#0 0.960 cannot delete non-empty directory: scripts
#0 0.960 deleting docs/
...
```

File: Dockerfile [hunk 1]
```diff
@@ -15,71 +18,84 @@ RUN apt-get update \
...
+# enforce perms
 USER node
 COPY --chown=node . .
 
 # make sure java is installed properly
 RUN java -version
 RUN javac -version
 
+# copy all package.json again
+RUN --mount=type=bind,target=/docker-context \
```
Comment (julien51) [inline Dockerfile:75]: Any reason we copy then again? Arguably since we copied all the files again (line 68), they are in there?

**Review by julien51 — APPROVED** [review]
I tested locally and when I remove the 2nd rsync things are working well... and caching means the 2nd build is MUCH faster!

**Comment by clemsos** [comment]
LG, lets try a few CI runs than we can merge

**Pull request merged**